src/main/java/org/radarcns/webapp/filter/AuthenticationFilter.java

```diff
@@ -0,0 +1,92 @@
+package org.radarcns.webapp.filter;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.Locale;
+import javax.annotation.Priority;
+import javax.ws.rs.container.ContainerRequestContext;
+import javax.ws.rs.container.ContainerRequestFilter;
+import javax.ws.rs.core.HttpHeaders;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.core.Response.Status;
+import javax.ws.rs.ext.Provider;
+import org.radarcns.auth.RadarSecurityContext;
+import org.radarcns.auth.authentication.TokenValidator;
+import org.radarcns.auth.config.ServerConfig;
+import org.radarcns.auth.config.YamlServerConfig;
+import org.radarcns.auth.exception.TokenValidationException;
+import org.radarcns.auth.token.RadarToken;
+import org.radarcns.config.Properties;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Checks the presence and validity of a Management Portal JWT. Will return a 401 HTTP status code
+ * otherwise.
+ */
+@Provider
+@Priority(1000)
+public class AuthenticationFilter implements ContainerRequestFilter {
+    private static final Logger logger = LoggerFactory.getLogger(AuthenticationFilter.class);
+    private static TokenValidator validator;
+
+    @Override
+    public void filter(ContainerRequestContext requestContext) {
+        String token = getToken(requestContext);
+        if (token == null) {
+            logger.warn("[401] {}: No token bearer header provided in the request",
+                    requestContext.getUriInfo().getPath());
+            requestContext.abortWith(
+                    Response.status(Status.UNAUTHORIZED)
```

Comment:
would it be possible to add a message as well in the response, to differentiate between e.g. unauthorized token (invalid scope) and token not available which would have 401?

Reply:
Done, it now has the `token_missing` and `token_invalid` errors with a message.